inc/calendar.php

```diff
@@ -297,3 +301,208 @@ function responsive_calendar_body_classes( $classes ) {
 }
 
 add_filter( 'body_class', 'responsive_calendar_body_classes' );
+
+if ( ! function_exists( 'responsive_calendar_get_calendar_id' ) ) {
+	/**
+	 * Retrieve the calendar ID for the calendar template.
+	 *
+	 * @return int $calendar_id Set from URL parameter if exists, else a site option.
+	 */
+	function responsive_calendar_get_calendar_id() {
+		return array_key_exists( 'cid', $_GET ) ? intval( $_GET['cid'] ) : get_option( 'bu_calendar_id' );
+	}
+}
+
+if ( ! function_exists( 'responsive_calendar_get_calendar_url' ) ) {
+	/**
+	 * Retrieve the calendar url for the calendar template.
+	 *
+	 * NOTE: This function may not be necessary. Seems like you could just call `get_permalink()`
```

Comment:
Would there be a situation where the calendar wouldn't be in the loop?  If not, and we are looking for the current post permalink, I'd say replace the above with `get_permalink()`

Reply:
Its called out of the loop in both scenarios its used on

- The calendar template partial (to help build event links by building on the current permalink with query args that represent single event)
- The sidebar args defined in `/inc/calendar.php`

I am removing the global $post variable regardless though, since calling `get_permalink` should refer to the main query's post as long as all other secondary queries honor `wp_reset_postdata()`.

So for now this function simply returns `get_permalink()`, and its only called twice, but in the future it may need to get refactored to retrieve the calendar permalink by a site option that stores the calendar template's post id, for example if a developer needed to provide a link to the calendar template from another page or post they could just do `get_permalink( get_option( 'bu_calendar_page_id' ) )` or something like that.